src/components/Table/TableHead.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { visuallyHidden } from '@mui/utils';
+import type { AutoUIEntityPropertyDefinition } from '../../AutoUI';
+import type { CheckedState, Order } from './utils';
+import { Material } from '@balena/ui-shared-components';
+
+const {
+	Box,
+	TableCell,
+	TableHead: MaterialTableHead,
+	TableRow,
+	TableSortLabel,
+	Checkbox,
+} = Material;
+
+interface TableHeadProps<T> {
+	columns: Array<AutoUIEntityPropertyDefinition<T>>;
+	numSelected?: number;
+	checkedState?: CheckedState;
+	onRequestSort: (
+		event: React.MouseEvent<HTMLSpanElement, MouseEvent>,
+		column: AutoUIEntityPropertyDefinition<T>,
+	) => void;
+	onSelectAllClick?: (
+		selected: T[] | undefined,
+		allChecked: CheckedState,
+	) => void;
+	order: Order;
+	orderBy: string;
+	rowCount: number;
+}
+
+export const TableHead = <T extends object>({
+	columns,
+	onSelectAllClick,
+	order,
+	orderBy,
+	numSelected,
+	checkedState,
+	rowCount,
+	onRequestSort,
+}: TableHeadProps<T>) => {
+	return (
+		<MaterialTableHead>
+			<TableRow>
+				<TableCell data-table="table_cel__sticky_header" padding="checkbox">
+					<Checkbox
+						color="primary"
+						// some does not seems to work as expected
```

Comment:
?

Reply:
🤦 forgot about that, it's a behaviour that I don't understand, on it